### DISCUSSION TITLE
Need help with record equality comparison

### DISCUSSION THREAD

**Discussion by FLAMESpl** [dotnet/csharplang#5978] [Q&A] [answered]
I have following xUnit tests:

```csharp
using System;
using Xunit;

namespace RecordEqualityTest;

public record Entity(Guid Id)
{
    public virtual bool Equals(Entity? other) => other?.Id == Id;
}

public record EntityWithCode(Guid Id, string Code) : Entity(Id)
{
    public virtual bool Equals(EntityWithCode? other) => base.Equals(other);
}

public class UnitTest1
{
    private static readonly Guid id = Guid.NewGuid();
    private readonly Entity entity = new(id);
    private readonly EntityWithCode entityWithCode = new(id, "Test");

    [Fact]
    public void Test1() => Assert.True(entity.Equals(entityWithCode));

    [Fact]
    public void Test2() => Assert.True(entityWithCode.Equals(entity));

    [Fact]
    public void Test3() => Assert.True(entity == entityWithCode);

    [Fact]
    public void Test4() => Assert.True(entityWithCode == entity);
}
```

I would like to have all tests passing, however test 2 and 4 fails. It turns out that when `EntityWithCode` is left hand argument then `Equals` method from base class is not called, because parent record has generated member `bool Equals(Entity)` that hides one from base record, what it does is calling `Equals(EntityWithCode)` with null argument.

I tried overriding `EntityWithCode.Equals(Entity)` like `EntityWithCode.Equals(EntityWithCode)` does but it won't compile with message "member with the same signature is already declared". 

How do I make proper equality comparison here?

**Comment by 333fred** [comment, marked as answer]
You don't use a record. See Andy's answer here: https://github.com/dotnet/csharplang/discussions/5586#discussioncomment-1903908.
